extensions/oplogPopulator/OplogPopulator.js

```diff
@@ -242,35 +242,61 @@ class OplogPopulator {
         this._changeStreamWrapper.start();
     }
 
+    _isPipelineImmutable() {
```

Comment:
nit: there may be more than one pipeline, and every one of them is immutable

```suggestion
    _arePipelinesImmutable() {
```